awx/main/tests/unit/tasks/test_jobs.py

```
@@ -427,3 +427,115 @@ def test_populate_claims_for_adhoc_command(workload_attrs, expected_claims):
+@mock.patch('awx.main.tasks.jobs.flag_enabled')
+def test_request_workload_identity_token_feature_flag_disabled(mock_flag_enabled, mock_get_client):
+    """Test that JWT request is skipped when feature flag is disabled."""
+    # Disable the feature flag
+    mock_flag_enabled.return_value = False
+    
+    # Create a test job
+    job = Job(id=789, name='Test Job')
+    
+    # Create a BaseTask instance and call pre_run_hook
+    task = jobs.BaseTask()
+    task.pre_run_hook(job, '/tmp/private_data')
```

Ruff (0.15.0)

[error] Probable insecure usage of temporary file or directory: "/tmp/private_data"

(S108)